components/director/pkg/jsonschema/validator.go

```diff
@@ -25,8 +26,26 @@ func NewValidator(jsonSchema string) (*validator, error) {
 	}, nil
 }
 
-func (v *validator) Validate(json string) (bool, error) {
+func NewValidatorFromSchema(jsonSchema interface{}) (*validator, error) {
```

Comment:
Please rename to `NewValidatorFromRawSchema`. But to be honest we won't use `NewValidatorFromStringSchema`, so probably it's safe to delete and then rename the constructor simply as `NewValidator`.